File: lib/Fetchers/base/Posts.js
import Fetcher from '../base/Fetcher';

export default class Posts extends Fetcher {
  static cacheObjectType = 'posts';

  /**
   * @param {String} board
   * @param {String} thread
   * @returns {Promise<*>}
   */
  static async get(board, thread) {
    return await this._get({ board, thread });
  }

  /**
   * @param {String} board
   * @param {String} thread
   * @returns {Promise<*>}
   */
  static async refreshCached(board, thread) {
    return await this._refreshCached({ board, thread });
  }
}

File: v2/app/routes/api/fourChan/info/posts.go
package info

import (
	"github.com/gofiber/fiber/v2"

	requesttimer "github.com/Allypost/thread-puller/app/providers/requestTimer"
	"github.com/Allypost/thread-puller/app/src/fourChan/fetcher"
	"github.com/Allypost/thread-puller/app/util/api/response"
)

func Posts(ctx *fiber.Ctx) error {
	board := ctx.Params("board")
	thread := ctx.Params("thread")

	timer := ctx.Locals("timer").(*requesttimer.RequestTimer)
	timer.Start("fetch")
	resp, err := fetcher.PostFetcher().Fetch(board, thread)
	timer.End("fetch")

	if err != nil {
		return response.SendError(ctx, fiber.StatusInternalServerError, "Error fetching boards", err)
	}

	return response.SendSuccess(ctx, resp)
}
